context.go

```diff
@@ -16,14 +16,41 @@ import (
 	"github.com/sdboyer/gps"
 )
 
+type BuildContext interface {
```

Comment:
Yep, `BuildContext` doesn't seem right. I don't love how `DepContext` stutters the package name (`dep.DepContext`). Is just `Context` okay? That's also pretty vague.

Most of the operations here are filesystem operations, apart from `SourceManager()`. Is there something we could call it related to that? I'm not sure.